File: src/mysqlcookbook/recipes/api/01_connect/connect4.py
#!/usr/bin/python3
# connect4.py: connect to the MySQL server specifying character set

import mysql.connector

try:
#@ _FRAG_
  conn_params = {
    "database": "cookbook",
    "host": "localhost",
    "user": "cbuser",
    "password": "cbpass",
    "charset": "utf8",
  }
#@ _FRAG_
  conn = mysql.connector.connect(**conn_params)
  print("Connected")
except:
  print("Cannot connect to server")
else:
  conn.close()
  print("Disconnected")

File: src/mysqlcookbook/recipes/api/01_connect/connect4.php
<?php
# connect4.php: connect to the MySQL server, specifying character set

try
{
#@ _FRAG_
  $dsn = "mysql:host=localhost;dbname=cookbook;charset=utf8";
#@ _FRAG_
  $dbh = new PDO ($dsn, "cbuser", "cbpass");
  print ("Connected\n");
}
catch (PDOException $e)
{
  die ("Cannot connect to server\n");
}
$dbh = NULL;
print ("Disconnected\n");
?>
